4788/src/main/cpp/Strategy/ShooterStrategy.cpp

```diff
@@ -0,0 +1,26 @@
+#include "Strategy/ShooterStrategy.h"
+#include <iostream>
+
+ShooterManualStrategy::ShooterManualStrategy(std::string name, Shooter &shooter, Controllers &contGroup) : Strategy(name), _shooter(shooter), _contGroup(contGroup) {
+  SetCanBeInterrupted(true);
+  SetCanBeReused(true);
+  Requires(&shooter);
+
+}
+
+void ShooterManualStrategy::OnUpdate(double dt) {
+  double manualFlyWheelPower = fabs(_contGroup.Get(ControlMap::manualFlyWheel)) > fabs(ControlMap::xboxDeadzone) ? _contGroup.Get(ControlMap::manualFlyWheel) : 0;
+
+  if (_contGroup.Get(ControlMap::innerCircleShoot)) {
+    _shooter.setAuto(200, dt);
+  } else if (_contGroup.Get(ControlMap::outerCircleShoot)) {
+    _shooter.setAuto(400, dt);
+  } else if (_contGroup.Get(ControlMap::shooterEject)) {
+    _shooter.setFlywheel(-ControlMap::shooterEjectPower);
+  } else {
+    _shooter.setFlywheel(manualFlyWheelPower);
+  }
+
+  double indexSpeed = fabs(_contGroup.Get(ControlMap::indexSpin)) > fabs(0.2) ? _contGroup.Get(ControlMap::indexSpin) : 0;
```

Comment:
Indexer can probably use constant values to prevent reliance on driver dexterity - remember, reliability is reproducability, which human drivers are historically pretty bad at. Best practice would be to use a sensor to detect when a ball needs to be indexed and where it is in the center of the bot, automating this process.

A driver should only have to push one button to shoot - any more is too much and decreases your cycle time.